### DISCUSSION TITLE
How do I use special characters in passwords?

### DISCUSSION THREAD

**Discussion by agezco** [rabbitmq/rabbitmq-server#12333] [Other] [answered]
### Describe the bug

Special characters are not accepted in passwords when adding them in console

### Reproduction steps

1. rabbitmqctl add_user test P@$$w0rD
2. rabbitmqctl set_permissions -p / test "." "." ".*"
3. rabbitmqctl set_user_tags test administrator
4. try to login (i tested with managment interface)

### Expected behavior

it should login, but in management interface i have massage: "Not_Authorized"

### Additional context

after adding user with another (simple) password, i was able to change password to  P@$$w0rD with management interface and it works after this

**Comment by michaelklishin** [comment, marked as answer]
@agezco if they are not accepted, how come Team RabbitMQ uses them in Tanzu RabbitMQ, the K8S Operator and there is a [dedicated doc section on these characters](https://www.rabbitmq.com/docs/access-control#passwords-and-shell-escaping)?

In addition to [string value escaping](https://www.rabbitmq.com/docs/next/configure#config-file) coming in 4.0 for those who specifically try to use `#` as one of the characters.

What you are looking at is your shell escaping or interpreting parts of the password before the CLI command is even executed.

**Comment by lukebakken** [comment]
@agezco basically, your question is "how does my shell work?"

Run this command:

```
echo rabbitmqctl add_user test P@$$w0rD
```
